src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -1,25 +1,121 @@
 package com.ericsson.gerrit.plugins.eiffel.messaging;
 
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.eiffelcommons.utils.HttpRequest;
+import com.ericsson.eiffelcommons.utils.ResponseEntity;
 import com.ericsson.gerrit.plugins.eiffel.configuration.EiffelPluginConfiguration;
 import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.EiffelEventSenderException;
+import com.ericsson.eiffelcommons.utils.HttpRequest.HttpMethod;
 import com.google.gson.Gson;
-import com.google.gson.JsonObject;
 
 public class EiffelEventSender {
+    private static final String GENERATE_PUBLISH_ENDPOINT = "/generateAndPublish/";
+    private static final String MESSAGE_PROTOCOL = "mp";
+    private static final String MESSAGE_TYPE = "msgType";
+    private static final String EIFFEL_PROTOCOL = "eiffelsemantics";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EiffelEventSender.class);
 
     private final Gson gson = new Gson();
-    private final EiffelPluginConfiguration pluginConfig;
+    private String eiffelMessage;
+    private String eiffelType;
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
 
     public EiffelEventSender(EiffelPluginConfiguration pluginConfig) {
+        this.httpRequest = new HttpRequest(HttpMethod.POST);
         this.pluginConfig = pluginConfig;
```

Comment:
When having multiple constructors with variable number of parameters I thought it was standard to call the other with the missing parameter i.e.

```
EiffelEventSender(pluginConfig, httprequest)
```

Reply:
Yes that's true, we have done so before.

Reply:
The first constructor now calls the second one.